### PR TITLE
Feature/global init with hoc

### PR DESCRIPTION
**Pull request by bluebill1049**
image

### CODE REVIEW

File: example/src/App.js [hunk 1]
```diff
@@ -12,9 +12,11 @@ import image3 from './wallpapers/Mountain-Range.jpg';
 import image4 from './wallpapers/Pink-Forest.jpg';
 import image5 from './wallpapers/Snow.jpg';
 import image6 from './wallpapers/Lion.jpg';
-import { SimpleImg, SimpleImgProvider } from 'react-simple-img';
+import { SimpleImg, init  } from './src/index';
 import './App.css';
 
+init();
```
Comment (yusinto) [inline example/src/App.js]: I think its better to call `init()` in `index.js` since it's a once-off global step.

File: src/logic/filterSrcset.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import parseSrcset from '../utils/parseSrcset';
 export const findClosestDpr = (result: Array<Object>, target: number) =>
```
Comment (yusinto) [inline src/logic/filterSrcset.js]: Should this file be moved alongside other utils under `utils` dir?

File: src/init.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  /* eslint-disable */
+  if (!window.IntersectionObserver) require('intersection-observer');
+  // $FlowIgnoreLine:
+  const observer = new IntersectionObserver(entries => onIntersection.call(this, entries), config);
+  return this ? observer : window.reactSimpleImgObserver = observer;
+  /* eslint-enable */
+}
+
+export function onIntersection(entries: Array<{ intersectionRatio: number, target: any }>) {
+  entries.forEach(({ intersectionRatio, target }) => (intersectionRatio > 0) && imageLoader.call(this, target));
```
Comment (yusinto) [inline src/init.js]: You need to break out of the for loop (you can't break with forEach so you'll have to change this to a plain for loop or use find) once you detect `intersectionRation > 0`.

File: src/simpleImgProvider.js [hunk 4]
```diff
@@ -2,7 +2,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import type { Context } from './simpleImg';
-import filterImgSrc from './logic/filterSrcset';
+import init from './init';
 
 export const APPEND_IMAGE_REF = '__ProgresssiveImagesAppendImageRef__';
```
Comment (yusinto) [inline src/simpleImgProvider.js]: Is it really neccessary to use dangling underscores to name these special context methods? It makes it really hard to read, both here and in SimpleImg component.

**Comment by bluebill1049** [comment]
Thanks @yusinto for reviewing! ❤️

**Comment by yusinto** [comment]
The example is broken. I tried `npm start` and I get:
```
./src/App.js
Module not found: Can't resolve './src/index'
```

File: README.md [hunk 5]
```diff
@@ -114,8 +107,33 @@ Image component working similar with standard `img` tag and with the following p
...
 | `width`        | number |          | image width apply to original image and placeholder                                                                                                                                                                |
 | `height`       | number |          | image height apply to original image and placeholder                                                                                                                                                               |
 | `alt`          | string |          |                                                                                                                                                                                                                    |
+
+## Advance Example
+
+Set up React Simple Img per page, you can use the following example without `init()`😘
+
+    import { SimpleImg, SimpleImgProvider } from './react-lazyLoad-images';
```
Comment (yusinto) [inline README.md]: from `'./react-lazyLoad-images';`? I think you mean `react-simple-img`?

File: src/index.js [hunk 6]
```diff
@@ -1,8 +1,10 @@
 import 'babel-polyfill';
 import SimpleImgProvider from './simpleImgProvider';
 import SimpleImg from './simpleImg';
+import init from './init';
 
 export {
+  init,
```
Comment (yusinto) [inline src/index.js]: Maybe rename `init` to `initialiseRsi` so it's more readable in the consumer's code.

**Comment by bluebill1049** [comment]
@yusinto example is not ready, i haven't published the code

File: src/index.js (hunk 6 above)
Comment (bluebill1049) [reply src/index.js]: how about `initSimpleImg`?

File: src/simpleImgProvider.js (hunk 4 above)
Comment (bluebill1049) [reply src/simpleImgProvider.js]: i followed egghead.io's tutorial from the paypal developer, maybe it's an over kill.

**Comment by bluebill1049** [comment]
@yusinto can't break in the for loop, because there could be more than 1 images in the viewport

**Pull request merged**